cli/src/torii/mod.rs

```diff
@@ -96,43 +95,57 @@ pub(crate) const fn query_status_code(query_error: &query::Error) -> StatusCode
 
 impl Reply for Error {
     fn into_response(self) -> Response {
-        const fn status_code(err: &Error) -> StatusCode {
-            use Error::*;
-            match err {
-                Query(e) => query_status_code(e),
-                VersionedTransaction(_)
-                | AcceptTransaction(_)
-                | RequestPendingTransactions(_)
-                | DecodeRequestPendingTransactions(_)
-                | EncodePendingTransactions(_)
-                | ConfigurationReload(_)
-                | TxTooBig => StatusCode::BAD_REQUEST,
-                Config(_) => StatusCode::NOT_FOUND,
-                PushIntoQueue(err) => match **err {
-                    queue::Error::Full => StatusCode::INTERNAL_SERVER_ERROR,
-                    queue::Error::SignatureCondition(_) => StatusCode::UNAUTHORIZED,
-                    _ => StatusCode::BAD_REQUEST,
-                },
-                #[cfg(feature = "telemetry")]
-                Prometheus(_) | Status(_) => StatusCode::INTERNAL_SERVER_ERROR,
+        use Error::*;
+        match self {
+            Query(err) => {
+                reply::with_status(utils::Scale(&err), query_status_code(&err)).into_response()
             }
+            // TODO Have a type-preserved response body instead of a stringified one #2279
+            VersionedTransaction(err)
+            | DecodeRequestPendingTransactions(err)
+            | EncodePendingTransactions(err) => {
+                reply::with_status(err.to_string(), err.status_code()).into_response()
+            }
+            _ => reply::with_status(Self::to_string(&self), self.status_code()).into_response(),
         }
+    }
+}
 
-        fn to_string(mut err: &dyn std::error::Error) -> String {
-            let mut s = "Error:\n".to_owned();
-            let mut idx = 0_i32;
+impl Error {
+    const fn status_code(&self) -> StatusCode {
+        use Error::*;
+        match self {
+            Query(e) => query_status_code(e),
+            VersionedTransaction(err)
+            | DecodeRequestPendingTransactions(err)
+            | EncodePendingTransactions(err) => err.status_code(),
+            AcceptTransaction(_)
+            | RequestPendingTransactions(_)
+            | ConfigurationReload(_)
+            | TxTooBig => StatusCode::BAD_REQUEST,
+            Config(_) => StatusCode::NOT_FOUND,
+            PushIntoQueue(err) => match **err {
+                queue::Error::Full => StatusCode::INTERNAL_SERVER_ERROR,
+                queue::Error::SignatureCondition(_) => StatusCode::UNAUTHORIZED,
+                _ => StatusCode::BAD_REQUEST,
+            },
+            #[cfg(feature = "telemetry")]
+            Prometheus(_) | Status(_) => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+
+    fn to_string(mut err: &dyn std::error::Error) -> String {
```

Comment:
Is it something different from [`ToString`](https://doc.rust-lang.org/std/string/trait.ToString.html)?

Reply:
```Rust
#[test]
fn test_to_string() {
    let err = Error::PushIntoQueue(Box::new(queue::Error::SignatureCondition(eyre!("msg"))));
    let str1 = Error::to_string(&err);
    let str2 = err.to_string();
    println!("{str1}");
    println!("{str2}");
}
```
`str1`:
```
Error:
    0: Failed to push into queue
    1: Failure during signature condition execution
    2: msg
```
`str2`:
```
Failed to push into queue
```
The equivalent of `str1` can be achieved by modifying `Display` impl, such as `#[error("Failed to push into queue: {0}")]`.
But I leave `Error::to_string` as it is because the string representation of Torii errors in general is off-topic of this PR and will be removed in #2279